server/src/main/java/org/candlepin/pinsetter/tasks/KingpinJob.java

```diff
@@ -152,20 +152,26 @@ public static JobStatus scheduleJob(JobCurator jobCurator,
         JobStatus status = null;
         try {
             status = jobCurator.create(new JobStatus(detail, trigger == null));
+            if (trigger != null) {
+                scheduler.scheduleJob(detail, trigger);
+            }
+            else {
+                scheduler.addJob(detail, false);
+            }
         }
         catch (EntityExistsException e) {
             // status exists, let's update it
             // in theory this should be the rare case
             status = jobCurator.find(detail.getKey().getName());
             jobCurator.merge(status);
         }
-
-        if (trigger != null) {
-            scheduler.scheduleJob(detail, trigger);
-        }
-        else {
-            scheduler.addJob(detail, false);
+        catch (RuntimeException e) {
```

Comment:
Should this catch Throwable? In what exception case would the job not be marked as failed?